nazgul/api.py

```diff
@@ -1,25 +1,58 @@
 from flask import Flask, render_template, request, Response, Blueprint
+from flask_httpauth import HTTPBasicAuth
+from werkzeug.security import generate_password_hash, check_password_hash
 
 from nazgul.mysql_model import MySQLModel, ModelError
 import nazgul.xmlrenderer as xmlrenderer
 import urllib.parse
-import os
+import os, time, logging
 
 bp = Blueprint("api", __name__, url_prefix="/api")
 hb = Blueprint("heartbeat", __name__)
 
+auth = HTTPBasicAuth()
+
+users = {"admin": generate_password_hash(os.environ.get("AUTH_PASS", "admin"))}
+
 test_db = os.environ.get("DATABASE_URL", "127.0.0.1")
 username = os.environ.get("DB_USER", "root")
 password = os.environ.get("DB_PASS", "")
 pool_size = int(os.environ.get("DB_CONNECTION_POOL_SIZE", 3))
 msm = MySQLModel(host=test_db, user=username, password=password, pool_size=pool_size)
 
+# Setup logging
+logger = logging.getLogger("nazgul")
+logger.setLevel(logging.DEBUG)
+
+fh = logging.FileHandler("nazgul.log")
+fh.setLevel(logging.DEBUG)
+
+logger.addHandler(fh)
+
+
+@auth.verify_password
+def verify_password(username, password):
+    if username in users:
+        # TODO: is current format for logs is TIME - IP ADDR - USER - ENDPOINT, is there more info tgat we need to log?
```

Comment:
We should start logging with mozlog JSON format in mind, but that can be done in a separate PR.